### PR TITLE
height/contentHeight accept function for dynamic value

### PR DESCRIPTION
**Pull request by aharpervc**
this makes it a lot easier to size the control to its container. for example, implementations could do `return container_el.clientHeight`

### CODE REVIEW

**Comment by aharpervc** [comment]
This is basically what I'm doing to solve https://github.com/fullcalendar/fullcalendar/issues/3039

**Comment by flick36** [comment]
+1

**Comment by arshaw** [comment]
i agree that #2704 should still be implemented, but this could be cool as well. I'd prefer to see it as the existing `height` or `contentHeight` option that simply accepts a function

**Comment by arshaw** [comment]
Philosophical question: how is this better than [calling the height setter method](http://fullcalendar.io/docs/display/height/)?

I'd rather keep the setter method as the only way to dynamically set an explicit pixel value, and then introduce a `height: 'parent'` option as described in https://github.com/fullcalendar/fullcalendar/issues/2704

**Comment by aharpervc** [comment]
It's a more elegant way to handle a dynamically changing container.

**Comment by arshaw** [comment]
@aharpervc can you provide hypothetical code samples to show how it's more elegant the existing technique of using the setter methods?

**Comment by aharpervc** [comment]
I'll give you my actual production code I'm running with this patch:

```javascript
this.timeline = $(container_el).fullCalendar(_.extend({
    timezone: "utc",
    editable: false,
    aspectRatio: 1.8,

    getContainerHeight: function() {
        return container_el.clientHeight - 15;
    },
    // etc...
```

There's no extra state to track and no extra events to hook up. Whenever calcSize is called internally, the current container client height is used. Now fullcalendar can vertically fill its container both on init and when the browser is resized. There's a `- 15` there because in my actual app, there's kind of a margin, but I didn't want to invent an extra container div to put fullcalendar inside of just for that.

Now, you could certainly re-implement this concept as a static property. However, you'd need to make assumptions on what is the appropriate container element (probably direct parent) and what is the appropriate margins (probably none). Those assumptions *may or may not be true* for all apps and *isn't relevant to the fullcalendar itself*. That's why it's more elegant to delegate the computation to the caller.

**Comment by arshaw** [comment]
ah ok, this is more clear to me now.

how is the height-update initiated? are you manually calling the [render method](http://fullcalendar.io/docs/display/render/) (which readjusts sizing) or are you merely waiting for the built-in window resize handler?

**Comment by aharpervc** [comment]
I'm not explicitly calling render, but also I do have `handleWindowResize` turned on.

**Comment by arshaw** [comment]
ok, this feature makes more sense to me now that i know you are leveraging window resize handling to adjust the height. thanks.

File: src/Calendar.js [hunk 1]
```diff
@@ -667,6 +667,10 @@ function Calendar_constructor(element, overrides) {
 	
 	
 	function _calcSize() { // assumes elementVisible
+		if (typeof options.getContainerHeight === 'function') {
+			suggestedViewHeight = options.getContainerHeight() - (headerElement ? headerElement.outerHeight(true) : 0);
+			return;
```
Comment (arshaw) [inline src/Calendar.js]: could you leverage the `else if` chain instead of doing a return?

**Comment by arshaw** [comment]
would you be able to write tests for this feature? mandatory to ensure it doesn't break accidentally in the future.

https://github.com/fullcalendar/fullcalendar/wiki/Automated-Tests
https://github.com/fullcalendar/fullcalendar/tree/master/tests/automated

please let me know if i can give you any help/advice.

**Comment by aharpervc** [comment]
I may be able to revisit this at some point, but I don't currently have your test environment set up, or much capacity to investigate it at the moment.

**Pull request merged**